src/lib/clipboard.ts

```diff
@@ -49,19 +49,31 @@ export class ClipboardHandler {
   };
 
   pasteExternal = async (e: ClipboardEvent): Promise<void> => {
-    for (const file of e.clipboardData.files) {
-      if (!file.type.includes("image")) continue;
-      const url = window.URL.createObjectURL(file);
-      fabric.Image.fromURL(url, async (obj: any) => {
-        await this.placeObject(obj);
-      });
-      return;
-    }
+    await this.processFiles(e.clipboardData.files);
     await this.paste();
   };
 
-  placeObject = async (obj: any): Promise<void> => {
-    const { x, y } = this.canvas.cursor;
+  processFiles = async (files: FileList, cursor?): Promise<void[]> => {
+    const promises = [...files]
```

Comment:
it is almost never the case i want to paste multiple pictures, and a lot of os-es don't support having multiple pictures in the clipboard even

Reply:
But all os-es support dragging multiple files. More importantly, if a clipboard were to contain multiple files, the first not being an image but a future one being an image, this handles that. This is also kind of what you were doing before, but then you just stopped the execution.

If you want to have it behave like stopping the execution for the clipboard paste, we can add an additional parameter `multiple: boolean` or `first: boolean` depending on which behavior is preferred by default. Then, the inner function performs a `find` instead of a `map`. I think this would be silly but your choice.

Reply:
wait, okay, so the previous case *already* handled the first one being an image and the rest being multiple images. though i agree i didn't consider the use case of dragging multiple files.